sdk/eventhub/azure-eventhubs/azure/eventhub/eventprocessor/_ownership_manager.py

```diff
@@ -0,0 +1,135 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# -----------------------------------------------------------------------------------
+
+import time
+import random
+import math
+from collections import Counter
+from azure.eventhub.aio import EventHubClient
+
+
+class OwnershipManager(object):
+    """Increases or decreases the number of partitions owned by an EventProcessor
+    so the number of owned partitions are balanced among multiple EventProcessors
+
+    An EventProcessor calls claim_ownership() of this class every x seconds,
+    where x is set by keyword argument "polling_interval" in EventProcessor,
+    to claim the ownership of partitions, create tasks for the claimed ownership, and cancel tasks that no longer belong
+    to the claimed ownership.
+
+    """
+    def __init__(self, event_processor, eventhub_client: EventHubClient, ownership_timeout: int):
+        self.all_parition_ids = []
+        self.eventhub_client = eventhub_client
+        self.eventhub_name = eventhub_client.eh_name
```

Comment:
Is there a specific reason that we make the eventhub_name an attribute on the instance, but that we also continue to get `eh_name` from the client instance later? Presumably, the values should stay the same....

Reply:
eventhub_name, consumer_group_name are two important attributes for an ownership. They usually appear together. They don't change anywhere in an ownership manager. Putting them together has better readability, I think.

Reply:
Let me rephrase, why are you *both* assigning the name to an attribute *and* accessing the name from `self.eventhub_client.eh_name` in other methods?

Reply:
That's a mistake. Changed all to self.eventhub_name